yaml.go

```diff
@@ -621,6 +625,9 @@ func getStructInfo(st reflect.Type) (*structInfo, error) {
 		}
 
 		if tag != "" {
+			if tag == "yaml_meta" && field.Type.String() != "yaml.StructMeta" {
```

Comment:
Do we need nesting?

Reply:
Good point we actually dont.

tag == "yaml_meta"  implies the other is true so we can move it out of the `tag != ""`